doc/developer-guides/hld/hv-hypercall.rst

```diff
@@ -3,19 +3,36 @@
 Hypercall / VHM upcall
 ######################
 
-HV currently supports hypercall APIs for VM management, I/O request
-distribution, and guest memory mapping.
+Hypercall/upcall used to request services between Guest VM and hypervisor,
+hypercall is from Guest VM to hypervisor, upcall is from hypervisor to Guest VM.
+Hypervisor currently supports hypercall APIs for VM management, I/O request
+distribution,interrupt injection, PCI assignment, guest memory mapping,
+power management and secure world switch.
 
-HV and Service OS (SOS) also use vector 0xF7, reserved as x86 platform
-IPI vector for HV notification to SOS. This upcall is necessary whenever
-there is device emulation requirement to SOS. The upcall vector 0xF7 is
-injected to SOS vCPU0
+There are some restrictions for hypercall and upcall:
+
+#. Only ring0 hypercalls from the guest VM are handled by the hypervisor,
+   otherwise hypervisor will inject GP to Guest VM.
+#. All the hypercalls (except secure world hypercalls) must be called from SOS VM,
+   otherwise hypervisor will inject UD to Guest VM.
+   see :ref:`secure-hypervisor-interface` for a detailed description.
+#. Need to protect the critical resoures such as global VM and VCPU strucutures
```

Comment:
Can we make it clear this protection should be done in guest or hypervisor or both?

Reply:
have updated and correct some typos